feat(menus): elgg_view_menu() now accepts an array of menu items

hypeJunction opened the pull request:
elgg_view_menu() now accepts an array of default menu items to add to the
unprepared menu bypassing menu registers. This is useful for rendering
multiple instances of the same menu on the page, e.g. filter

hypeJunction committed:
feat(menus): elgg_view_menu() now accepts an array of menu items

elgg_view_menu() now accepts an array of default menu items to add to the
unprepared menu bypassing menu registers. This is useful for rendering
multiple instances of the same menu on the page, e.g. filter

jdalsem commented:
Lgtm

mrclay commented on engine/lib/views.php:
```diff
@@ -739,6 +739,8 @@ function elgg_view_layout($layout_name, $vars = array()) {
  *
  *                          Options include:
+ *                              items => an array of unprepared menu items
+ *                                       as ElggMenuItem or menu item factory options
```
Can you clarify how this affects hooks?

hypeJunction replied:
Hook will receive menu items injected this way. I will update the doc

mrclay replied:
If folks want to avoid existing hook interference they can choose a unique menu name.

hypeJunction committed:
feat(menus): elgg_view_menu() can now render menus with custom views

elgg_view_menu() now accepts $vars["menu_view"] to render menus
using custom views. This can be helpful when a plugin uses a unique
menu name, but wants to preserve consistency in menu rendering

hypeJunction commented:
> If folks want to avoid existing hook interference they can choose a unique menu name.

This wasn't really possible. Added a second commit to allow custom view names.

mrclay reviewed (approved):
Great. Any docs we should add?

hypeJunction commented:
> Great. Any docs we should add?

Too lazy.

mrclay commented:
> Too lazy.

hackathoners must hack!
